rclpy/rclpy/duration.py

```diff
@@ -49,6 +49,21 @@ def __str__(self) -> str:
             return 'Infinite'
         return f'{self.nanoseconds} nanoseconds'
 
+    def __add__(self, other: 'Duration') -> 'Duration':
+        if isinstance(other, Duration):
+            return Duration(nanoseconds=other.nanoseconds + self.nanoseconds)
+        return NotImplemented
+
+    def __sub__(self, other: 'Duration') -> 'Duration':
+        if isinstance(other, Duration):
+            return Duration(nanoseconds=self.nanoseconds - other.nanoseconds)
+        return NotImplemented
+
+    def __mul__(self, other: int) -> 'Duration':
```

Comment:
should we take floating point here? so that user can do `Duration * 0.5`?

Reply:
I was a bit hesitant about the can of worms that floats tend to open up. So I assume we take a float and then recast it to an int?

Reply:
that is what rclcpp does, https://github.com/ros2/rclcpp/blob/rolling/rclcpp/src/rclcpp/duration.cpp#L225-L246

and IMO that is one of the use case. @clalancette @sloretz what do you thiunk?

Reply:
[4df59eb](https://github.com/ros2/rclpy/pull/1387/commits/4df59ebe1f210ed99dc588de4779bc1ef41c06f7)

Reply:
Yeah, agreed, I think we should make this API feel like the rclcpp one as much as possible.  So I'm for adding floats here.